src/Payum/Action/ResolveNextRouteAction.php

```diff
@@ -41,20 +39,6 @@ public function execute($request): void
 
     public function supports($request)
     {
-        if (
-            !$request instanceof ResolveNextRoute ||
-            !$request->getFirstModel() instanceof PaymentInterface
-        ) {
-            return false;
-        }
-
-        /** @var PaymentInterface $model */
-        $model = $request->getFirstModel();
-        /** @var PaymentMethodInterface $paymentMethod */
-        $paymentMethod = $model->getMethod();
-        /** @var GatewayConfigInterface $gatewayConfig */
-        $gatewayConfig = $paymentMethod->getGatewayConfig();
-
-        return $gatewayConfig->getFactoryName() === 'sylius.pay_pal';
```

Comment:
I know it's handled in the configuration, but maybe we should leave this check as it is? We cannot be *too much* secured and still somebody can override the service definition and change the tag 💃